src/processManager/library/AgentAttachmentRandomWalk.java

```diff
@@ -463,6 +473,14 @@ private void randomWalkAttachment(
 			
 			oldPoints = this.copyPoints(points);
 			
+			counter++;
+			
+			if (counter > 10)
+			{
+				if (this.attemptToReachBoundaryLayer(agent, points))
```

Comment:
This doesn't seem ideal. If we can't find something to attach to in 10 cycles we check another method that may return false if the agent evidently moves out of the top of the domain? But if it doesn't we continue to search for a place to attach to?